api/response/schema_query.py

```diff
@@ -94,49 +102,19 @@ def resolve_room_response(self, info, room_id):
             total_room_resources=total_room_resources,
             response=responses)
 
-    def get_all_reponses(self, info):
-        response = []
-        rooms = RoomModel.query.all()
-        for room in rooms:
-            room_name = room.name
-            room_response = Response.get_query(info).filter_by(room_id=room.id)
-            total_response = room_response.count()
-            all_responses, total_room_resources = Query.get_room_response(
-                self, room_response, room.id)
-            responses = RoomResponse(
-                room_id=room.id,
-                room_name=room_name,
-                total_responses=total_response,
-                total_room_resources=total_room_resources,
-                response=all_responses)
-            response.append(responses)
-        return response
-
-    def filter_rooms_by_responses(self, info, upper_limit, lower_limit):
-        all_responses = Query.get_all_reponses(self, info)
-        filtered_responses = []
-        for response in all_responses:
-            reponse_count = response.total_responses
-            if lower_limit <= reponse_count <= upper_limit:
-                filtered_responses.append(response)
-        return filtered_responses
-
-    def search_response_by_room(
-        self, info, upper_limit, lower_limit, room
-    ):
+    def search_response_by_room(self, info, upper_limit, lower_limit, room):
         search_result = []
         filtered_search = []
-        if upper_limit and lower_limit:
-            filtered_response = Query.filter_rooms_by_responses(
-                self, info, upper_limit, lower_limit)
-            for response in filtered_response:
-                if response.room_name.lower() == room.lower():
-                    filtered_search.append(response)
-            if filtered_search:
-                return filtered_search
-            else:
-                raise GraphQLError(
-                    "No response for this room at this range")
+        check_room_instance(
+                int, filter_rooms_by_responses, upper_limit=upper_limit,
```

Comment:
@RachaelNantale , I do not think we need to check if upper and lower limits are  integers since the graphene.Int() already takes care of this.